src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -49,6 +53,42 @@ public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessag
         /// </summary>
         protected string Namespace { get; private set; }
 
+        /// <summary>
+        /// Gets the unique identifier for this background job to distinguish this job instance in a multi-instance deployment.
+        /// </summary>
+        public string JobId { get; }
+
+        /// <summary>
+        /// Triggered when the application host is ready to start the service.
+        /// </summary>
+        /// <param name="cancellationToken">Indicates that the start process has been aborted.</param>
+        public override async Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (Settings.Options.IncludeTopicSubscription)
+            {
+                ServiceBusConnectionStringBuilder serviceBusConnectionString = await GetServiceBusConnectionStringAsync();
+
+                Logger.LogTrace("[Job: {JobId}] Creating subscription '{SubscriptionName}' on topic '{TopicPath}'...", JobId, _subscriptionName, serviceBusConnectionString.EntityPath);
+                var subscriptionDescription = new SubscriptionDescription(serviceBusConnectionString.EntityPath, _subscriptionName)
+                {
+                    AutoDeleteOnIdle = TimeSpan.FromHours(1),
+                    MaxDeliveryCount = 3,
+                    UserMetadata = $"Subscription created by Arcus job: '{JobId}' to process inbound CloudEvents."
+                };
+            
+                var ruleDescription = new RuleDescription("Accept-All", new TrueFilter());
+
+                var serviceBusClient = new ManagementClient(serviceBusConnectionString);
+                await serviceBusClient.CreateSubscriptionAsync(subscriptionDescription, ruleDescription, cancellationToken)
+                                      .ConfigureAwait(continueOnCapturedContext: false);
+
+                Logger.LogTrace("[Job: {JobId}] Subscription '{SubscriptionName}' created on topic '{TopicPath}'", JobId, _subscriptionName, serviceBusConnectionString.EntityPath);
+                await serviceBusClient.CloseAsync().ConfigureAwait(continueOnCapturedContext: false);
```

Comment:
We don't do this in other places nor it's a library, maybe best to remove for now then or not?